File: AdcosApi/Models/Cliente.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace AdcosApi.Models
{
    public class Cliente
    {
        [Key]
        public int Id { get; set; }
        [Required]
        [StringLength(150, ErrorMessage="O nome deve ter entre 1 e 150 caracteres")]
        public string Nome { get; set; }
        [StringLength(20, ErrorMessage="O documento deve ter entre 1 e 20 caracteres")]
        public string Documento { get; set; }
        public DateTime DataNascimento { get; set; }
        public string Telefone { get; set; }
        public int EnderecoId { get; set; }
        public Endereco Endereco { get; set; }
    }
}

File: angular-front/src/app/models/cliente.ts
import { Endereco } from './endereco';

export class Cliente {
  id: number;
  nome: string;
  documento: string;
  dataNascimento: string;
  telefone: string;
  endereco: Endereco;

  constructor(){
    this.endereco = new Endereco();
  }
}
